example/sofast_fringe/example_process_facet_ensemble.py

```diff
@@ -44,9 +46,11 @@ def example_driver():
     file_ensemble = join(sample_data_dir, 'Ensemble_lab_6x4.json')
 
     # Define save dir
-    dir_save = join(os.path.dirname(__file__), 'data/output/facet_ensemble')
-    if not os.path.exists(dir_save):
-        os.makedirs(dir_save)
+    dir_save = join(dirname(__file__), 'data/output/facet_ensemble')
+    ft.create_directories_if_necessary(dir_save)
+
+    # Set up logger
+    lt.logger(join(dir_save, 'log.txt'))

```

Comment:
It's typically better to do this step outside a function, such as in the `if __name__ == "__main__":` control, because lt.logger() should only be called once per process.

That being said, I think it's ok to leave it here, since the example_driver() function essentially _is_ the main process.

Reply:
Sounds good, I'll make sure to do that in the future.
I'll leave it here for the time being for the reason you mentioned. Also the save_dir needs to be set up before the log, so I'm fine with leaving it in this file.